lib/cs_guide_web/controllers/csv_controller.ex

```diff
@@ -0,0 +1,201 @@
+
+        "drinks" -> drinks_csv_content(conn)
+
+        "brands" -> brands_csv_content(conn)
+
+        "drink-types" -> drink_types_csv_content(conn)
+
+        "drink-styles" -> drink_styles_csv_content(conn)
+
+        "venue-types" -> venue_types_csv_content(conn)
+
+        _ ->  conn
+              |> put_status(:not_found)
+              |> put_view(CsGuideWeb.StaticPageView)
+              |> render("404.html")
+     end
+
+  end
+
+# VENUES
+
+  defp venues_csv_content(conn) do
+    venues = Venue.all() |> Enum.map(&venue_csv_data(&1))
+    |> CSV.encode(headers: [ :venue_name,
+                            :postcode,
+                            :phone_number,
+                            :cs_score,
+                            :description,
+                            :num_cocktails,
+                            :website,
+                            :address,
+                            :city,
+                            :twitter,
+                            :instagram,
+                            :facebook,
+                            :favourite,
+                            :lat,
+                            :long,
+                            :slug
+                            ])
+    |> Enum.to_list
+    |> to_string
+    conn
+
+    |> put_resp_content_type("text/csv")
+    |> put_resp_header("content-disposition", "attachment; filename=\"venues.csv\"")
+    |> send_resp(200, venues)
+  end
+
+  defp venue_csv_data(venue) do
+    %{
+      venue_name: venue.venue_name,
+      postcode: venue.postcode,
+      phone_number: venue.phone_number,
+      cs_score: venue.cs_score,
+      description: venue.description,
+      num_cocktails: venue.num_cocktails,
+      website: venue.website,
+      address: venue.address,
+      city: venue.city,
+      twitter: venue.twitter,
+      instagram: venue.instagram,
+      facebook: venue.facebook,
+      favourite: venue.favourite,
+      lat: venue.lat,
+      long: venue.long,
+      slug: venue.slug
+     }
+  end
+
+# Drinks
+
+  defp drinks_csv_content(conn) do
+    drinks = Drink.all() |> Enum.map(&drink_csv_data(&1))
+    |> CSV.encode(headers: [ :name,
+                             :abv,
+                             :description,
+                             :weighting,
+                             :ingredients
+                            ])
+    |> Enum.to_list
+    |> to_string
+
+    conn
+    |> put_resp_content_type("text/csv")
+    |> put_resp_header("content-disposition", "attachment; filename=\"drinks.csv\"")
+    |> send_resp(200, drinks)
+  end
+
+  defp drink_csv_data(drink) do
+    %{
+      name: drink.name,
+      abv: drink.abv,
+      description: drink.description,
+      weighting: drink.weighting,
+      ingredients: drink.ingredients
+     }
+  end
+
+# BRANDS
+  defp brands_csv_content(conn) do
+    brands = Brand.all() |> Enum.map(&brand_csv_data(&1))
+    |> CSV.encode(headers: [ :name,
+                             :description,
```

Comment:
@SimonLab can extract the list of fields/columns from the schema instead of duplicating it?